token/js/src/extensions/transferFee/instructions.ts

```diff
@@ -826,3 +826,166 @@ export function decodeHarvestWithheldTokensToMintInstructionUnchecked({
         },
     };
 }
+
+// SetTransferFee
+
+export interface SetTransferFeeInstructionData {
+    instruction: TokenInstruction.TransferFeeExtension;
+    transferFeeInstruction: TransferFeeInstruction.SetTransferFee;
+    transferFeeBasisPoints: number;
+    maximumFee: bigint;
+}
+
+export const setTransferFeeInstructionData = struct<SetTransferFeeInstructionData>([
+    u8('instruction'),
+    u8('transferFeeInstruction'),
+    u16('transferFeeBasisPoints'),
+    u64('maximumFee'),
+]);
+
+
+/**
+ * Construct a SetTransferFeeInstruction instruction
+ *
+ * @param mint              The token mint
+ * @param authority         The source account's owner/delegate
+ * @param signers           The signer account(s)
+ * @param transferFeeBasisPoints Amount of transfer collected as fees, expressed as basis points of the transfer amount
+ * @param maximumFee        Maximum fee assessed on transfers
+ * @param programID         SPL Token program account
+ *
+ * @return Instruction to add to a transaction
+ */
+export function createSetTransferFeeInstruction(
+    mint: PublicKey,
+    authority: PublicKey,
+    signers: (Signer | PublicKey)[] = [],
```

Comment:
I don't think the `= []` is doing anything here, and we also need to have at least one signer.